src/RemoteResourceS3Controller.js

```diff
@@ -17,16 +17,97 @@
+    if (u.pathname.charAt(u.pathname.length - 1) === '/') { //This is an S3 bucket
+      if (u.hostname.startsWith('s3.')) { //The bucket name is part of the path
+        let pathSegments = u.pathname.split('/');
+        pathSegments.shift(); //Discard the leading slash
+        u.pathname = pathSegments.shift();
+        u.search = `prefix=${pathSegments.join('/')}`;
+      } else { //The bucket name is part of the hostname
+        let hostnameSegments = u.hostname.split('.');
+        let bucket = hostnameSegments.shift();
+        u.hostname = hostnameSegments.join('.');
+        let prefix = u.pathname.slice(1, u.pathname.length);
+        u.search = `prefix=${prefix}`;
+        u.pathname = bucket;
+      }
+    }
+    return u.toString();
+  }
+
+  async _getBucketObjectRequestList(bucketRequest) {
+    const result = [];
+    const bucketUrl = objectPath.get(bucketRequest, 'options.url', objectPath.get(bucketRequest, 'options.uri'));
+    objectPath.get(bucketRequest, 'options.url',bucketUrl);
+    objectPath.del(bucketRequest, 'options.uri');
+    const url = this._fixUrl(bucketUrl);
+    bucketRequest.options.url=url;
+    const objectListResponse = await this.download(bucketRequest.options);
+    if (objectListResponse.statusCode >= 200 && objectListResponse.statusCode < 300) {
+      this.log.debug(`Download ${objectListResponse.statusCode} ${url}`);
+      const objectListString = objectListResponse.body;
+      const parser = new xml2js.Parser();
+      try {
+        const objectList = await parser.parseStringPromise(objectListString);
+        const xmlns= objectPath.get(objectList,'ListBucketResult.$.xmlns');
+        if(xmlns!=='http://s3.amazonaws.com/doc/2006-03-01/'){
+          this.log.warn(`Unexpected S3 bucket object list namespace of ${xmlns}.`);
+        }
+        let bucket = objectPath.get(objectList,'ListBucketResult.Name');
+        let objectsArray = objectPath.get(objectList,'ListBucketResult.Contents',[]);
+        objectsArray.forEach((o) => {
+          const objectKey = objectPath.get(o,'Key.0');
+          const reqClone = clone(bucketRequest);
+          const newUrl = new URL(url);
+          newUrl.pathname=`${bucket}/${objectKey}`;
+          newUrl.searchParams.delete('prefix');
+          reqClone.options.url=newUrl.toString();
+          result.push(reqClone);
+        });
+      }
+      catch(err) {
+        this.log.error(err, `Error getting bucket listing for ${url}`);
+      }
+    } else {
+      this.log.warn(`Download failed: ${objectListResponse.statusCode} | ${url}`);
+      return Promise.reject({ statusCode: objectListResponse.statusCode, uri: url });
+    }
+    return result;
+  }
+
+  async added() {
+    let requests = objectPath.get(this.data, ['object', 'spec', 'requests'], []);
+    let newRequests = [];
+    for(let i = 0; i<requests.length;i++){
+      let r = requests[i];
+      let url = objectPath.get(r, 'options.url');
+      if (url.endsWith('/')) {
```

Comment:
why both this line and https://github.com/razee-io/RemoteResourceS3/pull/18/files#diff-44b3a47ea97cfc5a9a2bb0bdf6c880f4R35?
also does this break if someone specifies a prefix in their URL?